src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -434,12 +443,23 @@ func isCsrApproved(csr *certificatesv1.CertificateSigningRequest) bool {
 	return false
 }
 
-func (c controller) PostInstallConfigs(ctx context.Context, wg *sync.WaitGroup) {
+func (c *controller) updateFinalizingProgress(ctx context.Context, stage models.FinalizingStage) {
+	_ = utils.RetryWithContext(ctx, math.MaxInt, GeneralWaitInterval, c.log, func() error {
+		return c.ic.UpdateFinalizingProgress(ctx, c.ClusterID, stage)
+	})
+}
+
+func (c *controller) waitOnStage(ctx context.Context, stage models.FinalizingStage, interval time.Duration, predicate func() bool) error {
```

Comment:
Found name not very obvious as we set stage here and wait on predicate. Not sure which name is better though